packages/visual-editor/src/components/puck/NearbyLocations.tsx

```diff
@@ -0,0 +1,339 @@
+import { ComponentConfig, Fields } from "@measured/puck";
+import {
+  resolveYextEntityField,
+  useDocument,
+  YextEntityField,
+  YextEntityFieldSelector,
+  Heading,
+  BackgroundStyle,
+  BasicSelector,
+  backgroundColors,
+  ThemeOptions,
+  HeadingLevel,
+  Section,
+  Phone,
+  fetchNearbyLocations,
+} from "../../index.js";
+import { useQuery } from "@tanstack/react-query";
+import { Address, Coordinate, HoursStatus } from "@yext/pages-components";
+import * as React from "react";
+
+export interface NearbyLocationsProps {
+  styles: {
+    backgroundColor?: BackgroundStyle;
+    cardBackgroundColor?: BackgroundStyle;
+  };
+  heading: {
+    level: HeadingLevel;
+    text: YextEntityField<string>;
+  };
+  cards: {
+    headingLevel: HeadingLevel;
+    phoneNumberFormat: "domestic" | "international";
+    phoneNumberLink: boolean;
+    hours: {
+      showCurrentStatus: boolean;
+      timeFormat?: "12h" | "24h";
+      dayOfWeekFormat?: "short" | "long";
+      showDayNames?: boolean;
+    };
+  };
+  coordinate: YextEntityField<Coordinate>;
+  radius: number;
+  limit: number;
+}
+
+const nearbyLocationsFields: Fields<NearbyLocationsProps> = {
+  styles: {
+    label: "Styles",
+    type: "object",
+    objectFields: {
+      backgroundColor: BasicSelector(
+        "Background Color",
+        ThemeOptions.BACKGROUND_COLOR
+      ),
+      cardBackgroundColor: BasicSelector(
+        "Card Background Color",
+        ThemeOptions.BACKGROUND_COLOR
+      ),
+    },
+  },
+  heading: {
+    label: "Section Heading",
+    type: "object",
+    objectFields: {
+      text: YextEntityFieldSelector({
+        label: "Text",
+        filter: {
+          types: ["type.string"],
+        },
+      }),
+      level: BasicSelector("Heading Level", ThemeOptions.HEADING_LEVEL),
+    },
+  },
+  cards: {
+    label: "Cards",
+    type: "object",
+    objectFields: {
+      headingLevel: BasicSelector("Heading Level", ThemeOptions.HEADING_LEVEL),
+      hours: {
+        label: "Hours",
+        type: "object",
+        objectFields: {
+          showCurrentStatus: {
+            type: "radio",
+            label: "Show Current Status",
+            options: [
+              { label: "Yes", value: true },
+              { label: "No", value: false },
+            ],
+          },
+          timeFormat: {
+            type: "radio",
+            label: "Time Format",
+            options: [
+              { label: "12-hour", value: "12h" },
+              { label: "24-hour", value: "24h" },
+            ],
+          },
+          showDayNames: {
+            type: "radio",
+            label: "Show Day Names",
+            options: [
+              { label: "Yes", value: true },
+              { label: "No", value: false },
+            ],
+          },
+          dayOfWeekFormat: BasicSelector("Day of Week Format", [
```

Comment:
Check with @apav-dev if this should be a selector or radio. Our other ones that have 3 or less options seem to be radio.